rules/concepts.md

```diff
@@ -194,8 +198,504 @@ Available types of conditions include:
+
+If there is an official helper library, it will be mentioned and used in the documentation of the rule engine, e.g. [JS Scripting](/addons/automation/jsscripting) with the [openhab-js](https://github.com/openhab/openhab-js) library.
+For unofficial helper libraries, you may search in the [community forum](https://community.openhab.org/search) for helper libraries for your rule language.
+An example for an unofficial helper library are [Ivan's helper libraries for Jython (community forum)](https://community.openhab.org/t/ivan-s-helper-libraries-oh3-python-javascript/116458).
+
+## Comprehensive Examples
+
+These following code snippets implement the examples from the top of this page using UI rules, [JS Scripting](https://openhab.org/addons/automation/jsscripting) and [DSL rules](/docs/configuration/rules-dsl).
+
+The code from the JS Scripting examples can be used in file-based scripts that are created inside the `/automation/js` folder and have `.js` as their file extension.
+You can paste the code snippets from the UI rules in the UI rule editor to inspect the examples.
+
+It is recommended to use helper libraries where they are available, as they provide an easy-to-use API to access openHAB functionality and avoid type problems related to the different languages combined.
+
+### When the sun rises, raise the blinds and adjust the temperature
+
+This example is using the [Astro Binding](https://www.openhab.org/addons/bindings/astro) which calculates many values for sun and moon, including sunrise.
+
+:::: tabs
+
+::: tab UI Rule
+
+![](./images/concepts/rule-sunrise.jpeg)
+
+The trigger in detail:
+
+![](./images/concepts/rule-sunrise-trigger.jpeg)
+
+The code:
+
+```yaml
+configuration: {}
+triggers:
+  - id: "1"
+    configuration:
+      event: START
+      channelUID: astro:sun:home:rise#event
+    type: core.ChannelEventTrigger
+conditions: []
+actions:
+  - inputs: {}
+    id: "2"
+    configuration:
+      itemName: gBlinds
+      command: UP
+    type: core.ItemCommandAction
+  - inputs: {}
+    id: "3"
+    configuration:
+      itemName: gThermostat
+      command: INCREASE
+    type: core.ItemCommandAction
+```
+
+:::
+
+::: tab DSL Rule
+
+```java
+rule "Raise the blinds & adjust temperature on sunrise"
+when
+  Channel 'astro:sun:home:rise#event' triggered START
+then
+  gBlinds.sendCommand(UP)
+  gThermostat.sendCommand(INCREASE)
+end
+```
+
+:::
+
+::: tab JS Scripting
+
+JS Scripting
+
+```javascript
+const { rules, triggers, items } = require('openhab');
+
+rules.JSRule({
+  name: 'Raise the blinds & adjust temperature on sunrise',
+  description: 'When the sun rises, raise the blinds and adjust the temperature',
+  triggers: [
+    triggers.ChannelEventTrigger('astro:sun:home:rise#event', 'START') // Triggers when the sun starts to rise
+  ],
+  execute: (event) => {
+      items.getItem('gBlinds').sendCommand('UP');
+      items.getItem('gTheromstat').sendCommand('INCREASE');
+  }
+});
+```
+
+:::
+
```

Comment:
::: tab jRuby

jRuby

```ruby
require 'openhab'

rule 'Raise the blinds & adjust temperature on sunrise' do
  channel 'astro:sun:home:rise#event'      
  run { 
    gBlinds << up
    gThermostat << increase
  }
end
```

:::

There's not enough easily accessible docs for the other languages. Those will take some time or we should ping people who might be able to submit a version for us after this gets merged upstream.

Reply:
Yeah, most languages don‘t have detailed docs in the addon documentation.
JS Scripting is a great example of how docs could (should) be, while most other automation addons only document some „Hello World“ logging.

Lets merge the concepts page upstream and then ping the right people.

Reply:
Missing event type `START` for the JRuby trigger?